.github/workflows/lint_code.yaml

```diff
@@ -102,9 +102,20 @@ jobs:
         run: |
           echo '{"array": ["dummy"], "variable": "dummy"}' > .mustache_config.json
           for yamlfile in $(find . -name "*.yml" -o -name "*.yaml"); do
-            sed "s|{{\([/#]\)[^}]*}}|{{\1array}}|g" $yamlfile > expanded.tmp  # replace mustache arrays
-            sed -i "s|{{[^#/].\{1,\}}}|{{variable}}|g" expanded.tmp           # replace mustache variables
-            mustache .mustache_config.json expanded.tmp > $yamlfile           # perform mustache expansion overwriting original file
+
+            filename=$(basename -- "$yamlfile")
+            extension="${filename##*.}"
+            filename="${filename%.*}"
+            test_config="tests/mustache/$filename.config.json"
+
+            if [ -e "$test_config" ]; then
+              mustache $test_config $yamlfile > $yamlfile
+            else
+              sed "s|{{\([/#]\)[^}]*}}|{{\1array}}|g" $yamlfile > expanded.tmp  # replace mustache arrays
+              sed -i "s|{{[^#/].\{1,\}}}|{{variable}}|g" expanded.tmp           # replace mustache variables
+              mustache .mustache_config.json expanded.tmp > $yamlfile           # perform mustache expansion overwriting original file
+            fi
+
```

Comment:
Is there a better way to do this? Can we configure yamllint to accept template statements or is there a mustache validations tool?

Reply:
It feels like there are two steps here: 1) expand templates with mustache and 2) validate YAML templates. The previous approach was to ignore (1) and try to strip the mustache syntax out but it looks like we've reached the limit of what's possible there.

Can we try adding a mustache config file with dummy values to the repo which will let us run something like:

```
for yamlfile in $(find . -name "*.yml" -o -name "*.yaml"); do
  mustache-expand $yamlfile | yamllint
done
```

Reply:
Probably best to push this to a new issue.